attest/verifier/types/src/verification.rs

```diff
@@ -309,8 +315,8 @@ impl EnclaveReportDataContents {
     }
 
     ///  Get the custom identity
-    pub fn custom_identity(&self) -> &[u8; 32] {
-        &self.custom_identity
+    pub fn custom_identity(&self) -> Option<&[u8; 32]> {
+        self.custom_identity.as_ref()
     }
 
     /// Returns a SHA256 hash of the contents of this structure.
```

Comment:
The choice of sha256 was more or less arbitrary, if desired we could remove this and leverage the `Digestible` implementation. As it feels like a bit of duplicate work determining what gets hashed/digested in both implementations.

At the same time I don't fully understand when one would prefer a `MerlinTranscript` over `sha256`.

Reply:
I think when trying to hash structs, the digestible stuff is considered safer since it also includes a domain separator, field names, lengths of arrays, etc.

Reply:
I created https://mobilecoin-eng.atlassian.net/browse/CORE-59 to re-visit consolidating the hash/digest usage